Pokedex/Main/MainTableViewController.swift

```diff
@@ -23,37 +23,31 @@ final class MainTableViewController: UITableViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         self.title = "Pokemons"
-        
-        // fetches the first 100 pokemons
-        client.fetchAllPokemons(limit: 100, offset: 0) { (result) in
-            if let firstPokemons = try? result.get() {
-                self.allPokemons = firstPokemons
-            }
-        }
+        bind()
+        viewModel.viewDidLoad()
     }
-    
 
+    func bind() {
+        viewModel.pokemonsOutput = { [weak self] pokemons in
+            self?.allPokemons = pokemons
+        }
+    }
 
     // MARK: - Table view data source
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return allPokemons?.results.count ?? 0
+        return allPokemons?.count ?? 0
```

Comment:
By having `[String]` instead of `[String]?`, you don't need to use nil coalescing and it's much more readable.